### PR TITLE
fixed to allow uesr to explore without login

### LINKED ISSUE #169: [Feature] User should be allow to browse without login
### Description

The user should be able to browse without login and while ordering it will ask for login that will enhance the user experience.

### Screenshots

_No response_

### Checklist

- [X] I have checked the existing [issues](https://github.com/pooranjoyb/popShop/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/pooranjoyb/popShop/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by Utsavladia**
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #169 

## Type of change

Please give a X on it which is applicable

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**_Test A Describe here_**
Tested by going to all the routes.

**_Test B Describe here (if Requred)_**

# Screenshorts and Vedios:

give screenshorts and vedio of the changes you made

# Checklist:
give a X on it which is applicable

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @Utsavladia! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by pooranjoyb** [comment]
Fix the build errors. Verify with `npm run build` @Utsavladia 
 
image

**Comment by pooranjoyb** [comment]
There are conflicts. Fix them. And lmk when are you free tomorrow, there are lot of issues that needs to be covered. @Utsavladia

**Comment by Utsavladia** [comment]
@pooranjoyb I have resolved the conflicts.
And I am free till 2pm.

**Comment by pooranjoyb** [comment]
Okay try to fix this now. @Utsavladia 

image

**Comment by Mahitej28** [comment]
> Okay try to fix this now. @Utsavladia
> 
> image

@Utsavladia try to work upon this, also there are merge conflicts take a pull from `dev` and then start working

**Comment by Utsavladia** [comment]
@Mahitej28 @pooranjoyb 
Fixed the Nav links in updated floating navbar and undefined section of dropdown.
image

**Comment by pooranjoyb** [comment]
Bruh, please remove use Conditional rendering for Login and Logout. Obviously how will a user Logout without Login?
@Utsavladia

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -111,21 +107,22 @@ function Navbar() {
...
             {showMenu && (
               <ul
                 tabIndex={0}
                 className="menu menu-sm dropdown-content mt-3 z-[1] p-2 bg-base-100 rounded-box w-52 shadow-2xl"
               >
                 <li onClick={handleCloseMenu}>
-                  <Link to={"/home/profile"} className="justify-between">
-                    {`${userName}` || <p>Profile</p>}
+                  <Link to={"/profile"} className="justify-between">
+                    {userName ? `${userName}` : <p>Login</p>}
```
Comment (pooranjoyb) [inline src/components/Navbar.tsx:118]: change this :/

```js
{username ? `${username}` : <p> Profile </p>}

```

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Comment by pooranjoyb** [comment]
Fix the conflicts. @Utsavladia

**Comment by Mahitej28** [comment]
@Utsavladia your pr has merge conflicts, resolve that and take a pull from `dev` before pushing commits.

**Comment by pooranjoyb** [comment]
Hii @Utsavladia please check the build errors. run `npm run build` in your local machine and debug accordingly. :)

**Comment by Utsavladia** [comment]
@pooranjoyb 
sure,
Resolved the conflicts and debugged the build error.

**Comment by Mahitej28** [comment]
@Utsavladia I can still see merge conflicts resolve it first

**Comment by Utsavladia** [comment]
@Mahitej28 
I hope it is fixed now.

**Comment by pooranjoyb** [comment]
merge conflicts :/ @Utsavladia

**Comment by pooranjoyb** [comment]
Closing as inactive.

**Comment by Utsavladia** [comment]
@pooranjoyb 
Sir, 
I kept resolving the conflicts again and again but due to delay in merging the branch new conflicts used to come and now also I could have resolved it I dont know how this notification got mixed.
sir I have spent time on this issue I request you to kindly let me fix and make it a successfull merge.

**Comment by pooranjoyb** [comment]
Okay @Mahitej28 & @AjayAsnani will check and respond accordingly.

**Comment by Mahitej28** [comment]
@Utsavladia will give you two days time to complete it.

**Comment by Utsavladia** [comment]
@pooranjoyb @Mahitej28 
Resolved the conflicts and synced with the latest branch I hope now it is working perfectly fine..

**Comment by Utsavladia** [comment]
@Mahitej28 
Mam this merge did not get counted in my contributions to GSSoC.
I think you will also have to assign this merge to me so that it gets counted 😊

**Comment by pooranjoyb** [comment]
> @Mahitej28 Mam this merge did not get counted in my contributions to GSSoC. I think you will also have to assign this pulll request to me so that it gets counted 😊

Wait for 24 hours, it takes time to update.

**Pull request merged**